src/App.tsx

```diff
@@ -1,12 +1,74 @@
-import React from 'react';
+import React, { useState } from 'react';
 import './App.scss';
+import { Good } from './react-app-env';
+import { getAll } from './api/goods';
+import { GoodsList } from './components/GoodsList';
 
-// import { getAll, get5First, getRed } from './api/goods';
-// or
-// import * as goodsAPI from './api/goods';
+const App: React.FC = () => {
+  const [goods, setGoods] = useState<Good[]>([]);
+  const [visibleGoods, setVisibleGoods] = useState<Good[]>([]);
 
-const App: React.FC = () => (
-  <h1>Dynamic list of Goods</h1>
-);
+  const goodsFromServer = () => {
+    getAll().then(allGoods => {
+      setGoods(allGoods);
+      setVisibleGoods(allGoods);
+    });
+  };
+
+  const cashedGoods = () => {
+    if (goods.length === 0) {
+      goodsFromServer();
+    }
+
+    setVisibleGoods(goods);
+  };
+
+  const firstFiveCashedGoods = () => {
+    if (goods.length === 0) {
+      goodsFromServer();
+    }
+
+    setVisibleGoods([...goods
+      .sort((a, b) => a.name.localeCompare(b.name))].slice(0, 6));
```

Comment:
].slice(0, 5));

Reply:
When I want to take first five elements, I should do it like `slice(0, 6)` , slice take all elements exsepts index 6)